app/scripts/actions/index.js

```diff
@@ -303,6 +329,34 @@ export const applyWorkflowToGranule = (granuleId, workflow) => ({
   }
 });
 
+export const applyRecoveryWorkflowToGranule = (granuleId) => {
+  return (dispatch) => {
+    return dispatch(getGranule(granuleId)).then((granuleResponse) => {
+      const { name, version } = collectionNameVersion(granuleResponse.data.collectionId);
+      return dispatch(
+        getCollection(name, version)
+      ).then((collectionResponse) => {
```

Comment:
It _might_ be more readable to abstract the logic for getting a collection by granule ID into a separate action that is used here

Reply:
Hmm... I think that's a good idea. I'll look into that.

Reply:
https://github.com/nasa/cumulus-dashboard/pull/573/commits/8ec2ae5d957121130c164f456bb7e379f892f6d8